home/tests.py

```diff
@@ -1,3 +1,81 @@
+            question = Question.objects.get(id=3)
+            ans1 = Answer(profile=profile, question=question, content='Answer 1', publish_date=datetime(2021, 4, 1,),
+                          likes_count=1, dislikes_count=0, is_edited=False)
+            ans2 = Answer(profile=profile, question=question, content='Answer 2', publish_date=datetime(2021, 4, 2,),
+                          likes_count=0, dislikes_count=0, is_edited=False)
+            ans1.save()
+            ans2.save()
+            return [ans1, ans2]
+
+        @pytest.mark.django_db
+        def test_thumb_up_answer(self, answers):
+            """
+            thumb_up_answer(answer_id) increase the thumbs up field of an answer by one
+            """
+            prev_thumb_val = answers[0].likes_count
+            answers[0].thumb_up_answer()
+            assert(prev_thumb_val == answers[0].likes_count - 1)
+
+        @pytest.mark.django_db
+        def test_thumb_down_answer(self, answers):
+            """
+            thumb_down_answer(answer_id) increase the thumbs down field of an answer by one
+            """
+            prev_thumb_val = answers[0].dislikes_count
+            answers[0].thumb_down_answer()
+            assert(prev_thumb_val == answers[0].dislikes_count - 1)
+
+        @pytest.mark.django_db
+        def test_set_is_edited(self, answers):
+            """
+            Tests functionality of set_is_edited
+            """
+            prev_is_edited_val = answers[0].is_edited
+            answers[0].set_is_edited(not prev_is_edited_val)
+            assert(prev_is_edited_val != answers[0].is_edited)
+
+        @pytest.mark.django_db
+        @pytest.mark.parametrize('filterType, expected', [('date', 'Answer 2'), ('votes', 'Answer 1')])
+        def test_answers_feed(self, filterType, expected, answers):
+            """
+            Tests if get_answers_feed returns queryset of all the answers of
+            question number 3 sorted by votes or answers
+            """
+            question = Question.objects.get(id=3)
+            answers_feed = question.get_answers_feed(filterType)
+            assert isinstance(answers_feed, QuerySet)
+            assert answers_feed[0].content == expected
+
+    class TestQuestionRelatedMethods:
+        @pytest.mark.django_db
+        def test_get_question_title(self):
+            """
+            get_question_title(question_id) returns a string for question
+            should return question subject and question title
+            """
+            question = Question.objects.get(id=1)
+            assert(question.get_question_title() == "Math-question from math course")
+
+    class TestHTMLRelated:
+        @pytest.mark.django_db
+        def test_display_question_page_url(self):
+            """
+            This test checks the returned status for routing to display-question feature
+            """
+            url = reverse('question-detail', args=[1])
+            c = Client()
```

Comment:
Use the `client` fixture that the `pytest-django` plugin provides for you. We've shown at least two examples of using it in class.

Also you should probably test more then just the HTTP status code, you should at least verify that the right template had bee used and the context passed to it contains the right contents

Reply:
OK fixed